src/Magento/FunctionalTestingFramework/Util/TestGenerator.php

```diff
@@ -2030,20 +2030,23 @@ private function resolveRuntimeReference($args, $regex, $func)
         $newArgs = [];
 
         foreach ($args as $key => $arg) {
+            $newArgs[$key] = $arg;
+
             preg_match_all($regex, $arg, $matches);
             if (!empty($matches[0])) {
-                $fullMatch = $matches[0][0];
-                $refVariable = $matches[1][0];
-                unset($matches);
-                $replacement = "{$func}(\"{$refVariable}\")";
+                foreach ($matches[0] as $matchKey => $fullMatch) {
+                    $refVariable = $matches[1][$matchKey];
+
+                    $replacement = $this->getReplacement($func, $refVariable);
+
+                    $outputArg = $this->processQuoteBreaks($fullMatch, $newArgs[$key], $replacement);
+                    $newArgs[$key] = $outputArg;
+                }
```

Comment:
Why this is changed to a loop?

Reply:
i think this is to accommodate resolving multiple {{_ENV.variable}} references for the fallback function `getReplacement`